### DISCUSSION TITLE
Support for man pages

### DISCUSSION THREAD

**Discussion by bharatvaj** [ScoopInstaller/Scoop#4921] [Ideas]
Can we support man pages? It would definitely make the command line experience much better in windows. I noticed that 'man' program is available in both 'unxutils' and 'busybox' packages but I'm not sure bringing in whole coreutils for man is a good idea. Also I don't know which paths are configured for these man programs. Is it possible to provide a standalone package for man? If so, how hard it would be to add man pages for available packages?

**Comment by rashil2000** [comment]
Support for man pages can probably done in the same way as completions: https://github.com/ScoopInstaller/Scoop/discussions/4803

But I don't think it is a very usable feature because:
- Windows doesn't really have the concept of man pages, MANPATH or INFOPATH env variables. Unix environments on Windows (i.e. Cygwin and MSYS2) ship with the "man" tool and man pages for all their packages, so they are readily usable there.
- Windows tools very rarely ship with man pages
- There's no standalone tool to view man pages (except busybox as you mentioned)


The alternative is to use something modern, like [cheatsheets](https://github.com/cheat/cheat) or [tldr-pages](https://github.com/tldr-pages/tldr)

**Comment by bharatvaj** [comment]
I didn't know about cheatsheets or tldr-pages before, thanks! they certainly will come handy. For my particular case I tend to use tools which are mostly cross-platform or standard unix tools and most of them provide man pages. Maybe If I submit a standalone package for man can this proposal be taken further? I tried cygwin and msys2 and both are very slow.

**Reply by rashil2000** [reply]
Cygwin and MSYS2 are the only options where you can run standard unix tools (except busybox, which doesn't have everything) - and man pages already work there. A standalone app for reading man pages would not really be useful because you won't have any pages in the first place.

**Reply by bharatvaj** [reply]
The uutils-coreutils package seems to provide decent amount of UNIX tools to Windows. If there is a framework for providing documentation, someone will provide man pages for these programs and also for other standard cross-platform programs like jq which are readily available somewhere on the internet. Standalone use case I can think of is something like the `libstdc++-doc` debian package which provides only the documentation of the c++ standard library.

**Reply by rashil2000** [reply]
> If there is a framework for providing documentation

This needs to exist first, even for Scoop to follow

**Reply by bharatvaj** [reply]
By framework I simply meant scoop providing support for *-doc packages and making man package available. What should exist in the place? If you meant the man program, I stumbled upon a port for windows. Sorry if I understood your statement wrong.

**Reply by rashil2000** [reply]
I meant a framework for consuming man-pages on Windows (outside of Cygwin) should exist, which Scoop can follow and adhere to. 

Scoop providing its own in-house implementation might not be the best idea.

**Reply by bharatvaj** [reply]
That makes sense, thanks for the clarification. Can I at least submit package for `mandoc` from https://embedeo.org/ws/doc/man_windows/mdocml-1.13.1-win32-embedeo-02.zip? As the site suggests, this doesn't provide the man program but offers a way to format the man pages and can be piped to `less` which seems to be the closest I can get to a man page experience on windows for now.

**Reply by rashil2000** [reply]
Packages in official buckets should come from trusted sources and/or preferably open source projects. Is there a way to determine if the binaries provided by embedeo.org are not malicious?

**Reply by bharatvaj** [reply]
I have no way determining that other than scanning it with an AV. I just ran VirusTotal on the binaries and the url and both came out to be safe.

**Reply by rashil2000** [reply]
Okay, please raise a PR

**Reply by bharatvaj** [reply]
Great! I'll do that.

**Comment by goyalyashpal** [comment]
> I meant a framework for consuming man-pages on Windows (outside of Cygwin) should exist, which Scoop can follow and adhere to.
> \- @ rashil2000 at https://github.com/ScoopInstaller/Scoop/discussions/4921#discussioncomment-2777097

hi! pipx provide man page support. and works flawlessly on windows with msys2. as windows doesnt really care about man pages, so, why would there be a standard for it?

and consequently, any implementation would follow to work with the msys2/cygwin etc.

so i think just thes 3 things would be required:
* a field in app manifest to allow apps to specify their man page location in the package they provide (similar to the fields `bin`, & `shortcuts`)
* a way to configure scoop about where to put man pages (by default - take inspiration from pipx)
* scoop to put & manage man pages at the above specified location

```console
$ pipx --help | grep -Ei "bin|manual"
Binaries can either be installed globally into isolated Virtual Environments
Symlinks to apps are placed in $USERPROFILE\.local\bin.
Symlinks to manual pages are placed in $USERPROFILE\.local\share\man.
  PIPX_BIN_DIR          Overrides location of app installations. Apps are
  PIPX_MAN_DIR          Overrides location of manual pages installations.
                        Manual pages are symlinked or copied here.

$ pipx environment | grep -Ei "bin|man" | tail -n2
PIPX_BIN_DIR=$USERPROFILE\.local\bin
PIPX_MAN_DIR=$USERPROFILE\.local\share\man

```

**Reply by goyalyashpal** [reply]
example is tools like `jq`. when i had installed it with msys2 (`pacboy debug -S jq:x`), there was a man page entry for that which was super helpful. after i migrated my installs to scoop, i am missing that.

**Reply by goyalyashpal** [reply]
other example being `aria2` downloader used by scoop itself. it refers towards its man page again and again at various places: 

```
# on interrupting a download
If there are any errors, then see the log file. See '-l' option in help/man page for details.

# on aria2c --help
Refer to man page for more information.
```

checking on msys2's pacman, `pacman -Ss aria2` sure enought, this is an application from linux world.
but as it's downloaded with scoop, i am left without man page for it.
